cmd/authd/daemon/migration_test.go

```diff
@@ -0,0 +1,109 @@
+package daemon
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMigrateOldCacheDir(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		oldDirExists     bool
+		newDirExists     bool
+		oldDirUnreadable bool
+		newDirUnreadable bool
+
+		wantErr          bool
+		wantOldDirExists bool
+		wantNewDirExists bool
+	}{
+		"Success if old dir does not exist": {oldDirExists: false, newDirExists: false},
+		"Success if old dir exists and new dir does not": {
+			oldDirExists:     true,
+			newDirExists:     false,
+			wantOldDirExists: false,
+			wantNewDirExists: true,
+		},
+		"Success if old dir exists and new dir exists": {
+			oldDirExists:     true,
+			newDirExists:     true,
+			wantOldDirExists: true,
+			wantNewDirExists: true,
+		},
+		"Success if old dir exists but is unreadable": {
+			oldDirExists:     true,
+			oldDirUnreadable: true,
+		},
+
+		"Error if new dir exists but is unreadable": {
+			oldDirExists:     true,
+			newDirExists:     true,
+			newDirUnreadable: true,
+			wantErr:          true,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			oldParentDir := t.TempDir()
+			newParentDir := t.TempDir()
+			oldDir := filepath.Join(oldParentDir, "cache")
+			newDir := filepath.Join(newParentDir, "cache")
+
+			if tc.oldDirExists {
+				err := os.Mkdir(oldDir, 0700)
+				require.NoError(t, err)
+
+				if tc.oldDirUnreadable {
```

Comment:
This can stay also in the main indentation no? We'd fail anyways if doesn't exist (so it's a badly written test), or can also do check for both statements.

Reply:
done